website/blog/2023-12-21-five-lessons-from-choosing-infrastructure-tooling/item.md

```diff
@@ -0,0 +1,161 @@
+    - authentication
+    - Authentik Security
+hide_table_of_contents: false
+image: ./tech-stack.png
+---
+
+> **_authentik is an open source Identity Provider that unifies your identity needs into a single platform, replacing Okta, Active Directory, and auth0. Authentik Security is a [public benefit company](https://github.com/OpenCoreVentures/ocv-public-benefit-company/blob/main/ocv-public-benefit-company-charter.md) building on top of the open source project._**
+
+---
+
+With great power (to choose your own tools) comes great responsibility. Not inheriting a legacy toolchain is an infrastructure engineer’s dream, but it can be hard to know where to start.
+
+As the first infrastructure engineer hired to work on authentik I saw the green field opportunities, but also the responsibility and long-term importance of choosing the best stack of tools and build processes. From my past roles, I already knew many of the considerations that we would need to look closely at.
+
+For example, we know that ease-of-maintenance is a primary consideration, as is the stability and probable longevity of the tool, how well the tools integrate, and of course the level of support we were likely to get for each tool.
+
+In this post we share what we’re using to build authentik, and the lessons behind those choices.
+
+![technology stack for startups](./tech-stack.png)
+
+<!--truncate-->
+
+## #1 Choices are often human, not technical
+
+If there isn’t much difference between two tools, the choice isn’t a technical decision. It’s going to come down to human factors like ease of use or the team’s familiarity with the tool. This is why we use [GitHub Actions](https://docs.github.com/en/actions) for our CI—[we’re already on GitHub](https://github.com/goauthentik) so it just makes sense.
+
+Familiarity with a tool means that you and your team can move faster, leading to higher business efficiency and a happier team.
+
+### We use Argo CD for GitOps
+
+When I joined Authentik Security, we were using [Flux CD](https://fluxcd.io/). Jens, our founder and CTO, had set up a small Kubernetes cluster to run an authentik instance for us to log into different services (some monitoring tools), and he was also deploying all of this using Flux CD.
+
+If you’re not familiar, Flux and [Argo CD](https://argo-cd.readthedocs.io/en/stable/) enable you to do GitOps: Whatever you want to deploy, you push that to a Git repository and then synchronize whatever is in production from that Git repository. Everything is committed and tracked in the Git history (helping you to understand what has changed and why).
+
+You also don’t need to do anything manually on the production servers or clusters—it’s all done in Git. This helps with auditing, as history is tracked, and you can easily find who made a change. And, since you don’t need to give access to your production servers and cluster to whoever is conducting the audit. They can see how everything is configured in the Git repo.
```

Comment:
```suggestion
You also don’t need to do anything manually on the production servers or clusters—it’s all done in Git. This helps with auditing, as history is tracked, and you can easily find who made a change. You don’t need to give access to your production servers and cluster to whoever is conducting the audit, since they can see how everything is configured in the Git repo.
```